common/app/templates/headerInlineJS/loadFonts.scala.js

```diff
@@ -1,87 +1,157 @@
-                    hinting = 'Auto';
-                } else {
-                    // All others use cleartype
-                    hinting = 'Cleartype';
-                }
-            }
-        }
-        return hinting;
-    }();
+    var ua = navigator.userAgent;
 
-    // Check to see if you should get webfonts, and then try to load them from localStorage if so
-    var cookieData = 'GU_fonts=off; domain=' + location.hostname + '; path=/';
+    // Determine what type of font-hinting we want.
+    var fontHinting = (function () {
+        var hinting;
+        function determineHinting() {
+            if (typeof hinting === 'undefined') {
+                try {
+                    var windowsNT = /Windows NT (\d\.\d+)/.exec(ua),
+                        version;
 
-    function disableFonts() {
-        document.cookie = cookieData + '; max-age=' + (60 * 60 * 24 * 365);
-    }
+                    hinting = 'Off';
+                    if (windowsNT) {
+                        version = parseFloat(windowsNT[1], 10);
 
-    function enableFonts() {
-        document.cookie = cookieData + '; expires=Thu, 01 Jan 1970 00:00:00 GMT';
-    }
-
-    function fontsEnabled() {
-        return document.cookie.indexOf('GU_fonts=off') === -1;
-    }
+                        // For Windows XP-7
+                        if (version >= 5.1 && version <= 6.1) {
+                            if (/Chrome/.exec(ua) && version < 6.0) {
+                                // Chrome on windows XP wants auto-hinting
+                                hinting = 'Auto';
+                            } else {
+                                // All others use cleartype
+                                hinting = 'Cleartype';
+                            }
+                        }
+                    }
+                } catch (e) {}
+            }
+            return hinting;
+        };
+        return determineHinting;
+    })();
 
     // Load fonts from `localStorage`.
     function loadFontsFromStorage() {
-        var storedFontFormat = localStorage['gu.fonts.format'],
-            fonts, fontFormat, font, dataAttrName, nameAndCacheKey, fontData;
-
-        if (storedFontFormat) {
-            fonts = document.querySelectorAll('.webfont');
-            fontFormat = JSON.parse(storedFontFormat).value;
-
-            for (var i = 0, j = fonts.length; i < j; ++i) {
-                font = fonts[i];
-                dataAttrName = 'data-cache-file-' + (fontHinting === 'Off' ? '' : 'hinted-' + fontHinting + '-') + fontFormat;
-                nameAndCacheKey = font.getAttribute(dataAttrName).match(/fonts\/([^/]*?)\/?([^/]*)\.(woff2|woff|tff).json$/);
-                fontData = localStorage['gu.fonts.' + nameAndCacheKey[2] + '.' + nameAndCacheKey[1]];
-
-                if (fontData) {
-                    font.innerHTML = JSON.parse(fontData).value;
-                    font.setAttribute('data-loaded-from', 'local');
+        if ("localStorage" in window) {
```

Comment:
The quick test shows that when you disable cookies

window['localStorage'] will throw Uncaught DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.

but ('localStorage' in window) will return true.

For example Modernizr is using .setItem/.getItem inside try/catch test

So I believe @sndrs is right but @piuccio as well. Theoretically the if test is redundant as we don't need to check for localStorage object presence (as it doesn't necessary means we can access it) but we need to test if we actually can use it.